src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -184,14 +195,25 @@ public ProCombatMoveAI(final ProUtils utils, final ProBattleUtils battleUtils,
     // Calculate attack routes and perform moves
     doMove(attackMap, moveDel, data, player, isSimulation);
 
+    // Set strafing territories to avoid retreats
+    final List<Territory> strafingTerritories = new ArrayList<Territory>();
+    for (final Territory t : attackMap.keySet()) {
+      if (attackMap.get(t).isStrafing()) {
+        strafingTerritories.add(t);
+      }
+    }
+    ai.setStoredStrafingTerritories(strafingTerritories);
```

Comment:
Good point. I think I did this since I'm planning to replace the attackMap with an actual object to manage territories (TerritoryManager or something). This functionality will then probably be moved into that and be called something like 'getStrafingTerritories'.